components/script/dom/htmllinkelement.rs

```diff
@@ -113,6 +113,9 @@ impl HTMLLinkElement {
             doc.remove_stylesheet(self.upcast(), s)
         }
         *self.stylesheet.borrow_mut() = Some(s.clone());
+        if let Some(cssom_stylesheet) = self.cssom_stylesheet.get() {
+            cssom_stylesheet.set_owner(None);
+        }
         self.cssom_stylesheet.set(None);
```

Comment:
I think it's good to move the cleanup of `cssom_stylesheet` into a function so that we can reuse it in both `set_stylesheet` and `unbind_from_tree`. (Same for `htmlstyleelement.rs`)

Ex. (`pseudo code`)

```rust
fn cssom_stylesheet_cleanup(&self) {
    if let Some(cssom_stylesheet) = self.cssom_stylesheet.get() {
        cssom_stylesheet.set_owner(None);
    }
    self.cssom_stylesheet.set(None);
}

fn set_stylesheet(...) {
    ...
    *self.stylesheet.borrow_mut() = Some(s.clone());
    self.cssom_stylesheet_cleanup();
    ...
}

fn unbind_from_tree(...) {
    ...
    if let Some(s) = self.stylesheet.borrow_mut().take() {
        self.cssom_stylesheet_cleanup();
        ...
    }
}
```